learning_resources_search/views.py

```diff
@@ -74,6 +80,77 @@ def get(self, request):
             return Response(errors, status=400)
 
 
+@method_decorator(blocked_ip_exempt, name="dispatch")
+@extend_schema_view(
+    get=extend_schema(
+        parameters=[LearningResourcesSearchRequestSerializer()],
+        responses=LearningResourceSearchResponseSerializer(),
+    ),
+)
+@action(
+    methods=["POST", "DELETE"],
+    detail=False,
+    name="Subscribe/Unsubscribe User to Search Query",
+)
+class SearchSubscriptionView(ESView):
+    """
+    Subscribe a user to a search query
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    @extend_schema(summary="Subscribe")
+    def post(self, request):
+        request_data = LearningResourcesSearchRequestSerializer(data=request.data)
+        if request_data.is_valid():
+            response = subscribe_user_to_search_query(
+                request.user, request_data.data | {"endpoint": LEARNING_RESOURCE}
+            )
+            return Response(response)
+        else:
+            errors = {}
+            for key, errors_obj in request_data.errors.items():
+                if isinstance(errors_obj, list):
+                    errors[key] = errors_obj
+                else:
+                    errors[key] = list(set(chain(*errors_obj.values())))
+            return Response(errors, status=400)
+
+    @extend_schema(summary="Subscribe")
+    def get(self, request):
+        request_data = LearningResourcesSearchRequestSerializer(data=request.GET)
+        if request_data.is_valid():
+            is_subscribed = user_subscribed_to_query(
+                request.user, request_data.data | {"endpoint": LEARNING_RESOURCE}
+            )
+            return Response({"is_subscribed": is_subscribed})
+        else:
+            errors = {}
+            for key, errors_obj in request_data.errors.items():
+                if isinstance(errors_obj, list):
+                    errors[key] = errors_obj
+                else:
+                    errors[key] = list(set(chain(*errors_obj.values())))
+            return Response(errors, status=400)
+
+    @extend_schema(summary="Unsubscribe")
+    def delete(self, request):
+        request_data = LearningResourcesSearchRequestSerializer(data=request.data)
```

Comment:
I think it may not be best API practice to send json data in a DELETE request, and in this case it's not really deleting the PercolateQuery, just removing the user from it.  

Could this be a distinct `unsubscribe` action/url instead with a url like `http://localhost:8063/api/v1/learning_resources_search_subscribe/unsubscribe`?


Will there ever be a need to get a list of all the user's search subscriptions?  In that case I think it might be better to have a more standard DRF viewset where:

GET `http://localhost:8063/api/v1/learning_resources_search_subscriptions` returns all the user's subscribed queries.  This could be used on the frontend to determine if the current query matches a subscribed query (and facilitate allowing the user to view/manage their saved queries)

POST `http://localhost:8063/api/v1/learning_resources_search_subscriptions` with JSON data upserts a PercolateQuery with the user as a member, very similar to how it currently works in this PR

POST `http://localhost:8063/api/v1/learning_resources_search_subscriptions/<id>/unsubscribe` (no json data) or `http://localhost:8063/api/v1/learning_resources_search_subscriptions/unsubscribe` (with json data) removes the user's subscription to the query specified by id (or json data)

Reply:
I do recall long ways back running into odd issues with client libraries not universally allowing for a body on delete methods.

 I've refactored this so that:

**GET** /api/v1/**learning_resources_user_subscription** -> lists percolate queries for user
**POST** /api/v1/learning_resources_user_subscription/**subscribe** -> subscribe user to query
**POST** /api/v1/learning_resources_user_subscription/**unsubscribe** -> unsubscribe user to query
**POST** /api/v1/learning_resources_user_subscription/**<pk>/unsubscribe** -> unsubscribe user by id